Datastore/tests/Unit/FilterTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Tests\Unit;
+
+use Google\Cloud\Datastore\Filter;
+use Google\Cloud\Datastore\Query\Query;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group datastore
+ * @group datastore-query
+ */
+class FilterTest extends TestCase
+{
+    /**
+     * @dataProvider getCompositeFilterCases
+     */
+    public function testCompositeFilterMethods($methodName, $filters)
+    {
+        $filter = call_user_func_array([Filter::class, $methodName], [$filters]);
+
+        $this->assertArrayHasKey('compositeFilter', $filter);
+        $compositeFilter = $filter['compositeFilter'];
+
+        $this->assertEquals($compositeFilter['filters'], $filters);
+
+        // Taking substring of $methodName from 3rd characted so that
+        // 'doAnd' => 'And' and 'doOr' => 'Or'.
+        $expectedOperator = substr($methodName, 2);
```

Comment:
Once you rename `doOr -> or` & `doAnd -> and`, probably this isn't needed.

Reply:
Yeah, thanks for pointing out. Fixed it.